src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java

```diff
@@ -222,8 +230,17 @@ public void relinkInitLaunchDir(Path pathToNucleusDistro, boolean linkCurrentToI
      */
     @SuppressFBWarnings(value = "NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE", justification = "Spotbugs false positive")
     public static Path locateCurrentKernelUnpackDir() throws IOException, URISyntaxException {
-        Path parentDir = new File(KernelAlternatives.class.getProtectionDomain().getCodeSource().getLocation()
-                .toURI()).toPath().getParent();
+#if ANDROID
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Syntax error on tokens, delete these tokens

This file contains syntax errors and can&#39;t be analyzed by all of CodeGuru Reviewer&#39;s detectors. Fix the syntax errors and run the repository analysis or submit the pull request again.